parsers/US_SPP.py

```diff
@@ -297,6 +328,109 @@ def fetch_wind_solar_forecasts(
     return data
 
 
+def fetch_live_exchange(
+    zone_key1: str,
+    zone_key2: str,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+
+    data = get_data(EXCHANGE_URL, session)
+
+    data = data.dropna(axis=0)
+    data["GMTTime"] = pd.to_datetime(data["GMTTime"], utc=True)
+    data = data.loc[data["GMTTime"] <= target_datetime]
+    data = data.set_index("GMTTime")
+
+    exchanges = format_exchange_data(
+        data=data, zone_key1=zone_key1, zone_key2=zone_key2, logger=logger
+    )
+    return exchanges
+
+
+def fetch_historical_exchange(
+    zone_key1: str,
+    zone_key2: str,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+
+    filename = target_datetime.strftime("TieFlows_%b%Y.csv")
+    file_url = HISTORICAL_EXCHANGE_URL + filename
```

Comment:
It's more safe to use either f strings or .format.

Reply:
f strings would be the preferred way as they are slightly more performant.